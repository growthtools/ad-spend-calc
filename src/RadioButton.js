import React from "react";
import PropTypes from "prop-types";

const RadioButton = ({ value, onChange }) => {
  return (
    <div class="radio-button-wrapper">
      <label class="radio-button-label">
        Radio Button Label
          <input type="radio" name="name" value="name"></input>
        <span class="radio-button-checkmark"></span>
      </label>
    </div>
  );
};

RadioButton.propTypes = {
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default RadioButton;

