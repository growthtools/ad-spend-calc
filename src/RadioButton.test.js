/* globals test expect */
import React from "react";
import { render } from "@testing-library/react";
import RadioButton from "./RadioButton";

const defaultProps = {
  id: "id",
  value: "first",
  options: ["first", "second"],
  onChange: () => {},
};

test("matches design when an item is selected", () => {
  const { asFragment } = render(<RadioButton {...defaultProps} />);
  expect(asFragment()).toMatchSnapshot();
});

test("matches design when an item is not selected", () => {
  const props = Object.assign([], defaultProps, {
    value: "",
  });

  const { asFragment } = render(<RadioButton {...props} />);
  expect(asFragment()).toMatchSnapshot();
});
