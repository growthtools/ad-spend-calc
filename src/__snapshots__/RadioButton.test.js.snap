// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches design when an item is not selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx ghEjMF"
  >
    <label
      class="container"
    >
      first
      <input
        name="id"
        type="radio"
        value="first"
      />
      <span
        class="checkmark"
      />
    </label>
  </div>
  <div
    class="sc-bdnylx ghEjMF"
  >
    <label
      class="container"
    >
      second
      <input
        name="id"
        type="radio"
        value="second"
      />
      <span
        class="checkmark"
      />
    </label>
  </div>
</DocumentFragment>
`;

exports[`matches design when an item is selected 1`] = `
<DocumentFragment>
  <div
    class="sc-bdnylx ghEjMF"
  >
    <label
      class="container checked"
    >
      first
      <input
        checked=""
        name="id"
        type="radio"
        value="first"
      />
      <span
        class="checkmark"
      />
    </label>
  </div>
  <div
    class="sc-bdnylx ghEjMF"
  >
    <label
      class="container"
    >
      second
      <input
        name="id"
        type="radio"
        value="second"
      />
      <span
        class="checkmark"
      />
    </label>
  </div>
</DocumentFragment>
`;
